packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -93,7 +93,211 @@ function addRefetchToLeafSegments(
 
   return appliedPatch
 }
-export function navigateReducer(
+
+// These implementations are expected to diverge significantly, so I've forked
+// the entire function. The one that's disabled should be dead code eliminated
+// because the check here is statically inlined at build time.
+export const navigateReducer = process.env.__NEXT_PPR
```

Comment:
wonder if it'd be useful to annotate the action type that we dispatch to devtools with something like `navigate (ppr)` just to sanity check which reducer is being used on the client. But I imagine it'll be pretty obvious which one is being used without that once more functionality is wired up, so maybe not worth the effort

Reply:
yeah you can just log in the reducer, or log process.env.__NEXT_PPR, since it's a static condition